lib/active_graph/node/has_n.rb

```diff
@@ -111,13 +111,17 @@ def init_cache
 
       def add_to_cache(object, rel = nil)
         (@cached_rels ||= []) << rel if rel
-        (@cached_result ||= []).tap { |results| results << object if object && !results.include?(object) }
+        (@cached_result ||= []).tap { |results| results << object if !results.include?(object) } if object
```

Comment:
There is a reason why I changed this code. After adding the `rel` function to AssociationProxy, I noticed that the following case is working incorrectly.

```
role = Role.find(xxx)
role.manager.source_rel
role.manager.source
```

Even though the manager has a source, if we call `source` after calling `source_rel`, it returns nil. Because we used the `add_to_cache` method in line `has_n.rb:118`, `@cached_result` was initialized as an empty array then in line `has_n.rb:81` we check `@cached_result`, since it is initialized we use it, i.e. the empty array. To avoid this, I made `@cached_result` initialize only when the object exists. All specs and manual tests passed successfully.

* Object parameter is nil only when we use the `add_to_cache` method for rels.